### PR TITLE
docs(cmd): add documentation for cmd package

### PR DESCRIPTION
**Pull request by cboin1996**
Changes:
- Added cmd-processes.md within the cmd package to describe each profile controller. 
- Updated README.md to include a link to the helm chart for deploying the controllers.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,9 +7,9 @@ defined with a CustomResourceDefinition (CRD).
 
 This particular example demonstrates how to perform basic operations such as:
 
-* How to register a new custom resource (custom resource type) of type `Foo` using a CustomResourceDefinition.
-* How to create/get/list instances of your new resource type `Foo`.
-* How to setup a controller on resource handling create/update/delete events.
+- How to register a new custom resource (custom resource type) of type `Foo` using a CustomResourceDefinition.
```
Comment (blairdrummond) [inline README.md:10]: Are the current resources of type `Foo`? :-P

File: README.md [hunk 2]
```diff
@@ -31,11 +24,64 @@ The profiles controller uses [client-go library](https://github.com/kubernetes/c
 The details of interaction points of the profiles controller with various mechanisms from this library are
 explained [here](docs/controller-client-go.md).
 
+## Controllers
+
+The [cmd](./cmd) package contains source files for a variety of profile controllers for Kubeflow.
+
+For more information about profile controllers, see the documentation for the [client-go](https://github.com/kubernetes/client-go/) library, which contains a variety of mechanisms for use when
+developing custom profile controllers. The mechanisms are defined in the
+[tools/cache folder](https://github.com/kubernetes/client-go/tree/master/tools/cache) of the library.
```
Comment (blairdrummond) [inline README.md:33]: These point to the wrong repos

File: README.md [hunk 3]
```diff
@@ -96,15 +142,20 @@ kubectl create -f artifacts/examples/example-foo.yaml
...
 
 CustomResourceDefinitions can be used to implement custom resource types for your Kubernetes cluster.
 These act like most other Resources in Kubernetes, and may be `kubectl apply`'d, etc.
 
 Some example use cases:
 
-* Provisioning/Management of external datastores/databases (eg. CloudSQL/RDS instances)
-* Higher level abstractions around Kubernetes primitives (eg. a single Resource to define an etcd cluster, backed by a Service and a ReplicationController)
+- Provisioning/Management of external datastores/databases (eg. CloudSQL/RDS instances)
+- Higher level abstractions around Kubernetes primitives (eg. a single Resource to define an etcd cluster, backed by a Service and a ReplicationController)
```
Comment (blairdrummond) [inline README.md:158]: Don't need this info

File: README.md [hunk 4]
```diff
@@ -113,7 +164,7 @@ In practice, this Spec is arbitrary key-value data that specifies the configurat
 
 For example, if you were implementing a custom resource for a Database, you might provide a DatabaseSpec like the following:
 
-``` go
+```go
 type DatabaseSpec struct {
 	Databases []string `json:"databases"`
 	Users     []User   `json:"users"`
```
Comment (blairdrummond) [inline README.md:170]: Don't need this

File: README.md [hunk 5]
```diff
@@ -154,11 +205,13 @@ kubectl create -f artifacts/examples/crd-status-subresource.yaml
...
+
 The [group](https://kubernetes.io/docs/reference/using-api/#api-groups) version of the custom resource in `crd.yaml` is `v1alpha`, this can be evolved to a stable API version, `v1`, using [CRD Versioning](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/).
 
 ## Cleanup
 
 You can clean up the created CustomResourceDefinition with:
+
 ```sh
 kubectl delete crd foos.profilescontroller.k8s.io
 ```
```
Comment (blairdrummond) [inline README.md:217]: Don't need this

File: README.md (hunk 2 above)
Comment (cboin1996) [reply README.md:33]: Isn't that where the source example for creating a profile controller is found? Are we not using the client-go library then?

Comment (blairdrummond) [reply README.md:33]: Sorry I read quickly

"For more information about profile controllers"

The upstream repo doesn't refer to `Profile` controller, just generic controllers of a Foo object

**Review by cboin1996 — COMMENTED** [review]
Follow up with the proper commit, and make changes that you havent already addressed.

File: README.md [hunk 6]
```diff
@@ -1,176 +1,98 @@
...
+For more information about profile controllers, see the documentation for the [client-go](https://github.com/kubernetes/client-go/) library, which contains a variety of mechanisms for use when
+developing custom profile controllers. The mechanisms are defined in the
+[tools/cache folder](https://github.com/kubernetes/client-go/tree/master/tools/cache) of the library.
 
-Some example use cases:
+### [authpolicy.go](./cmd/authpolicy.go)
 
-* Provisioning/Management of external datastores/databases (eg. CloudSQL/RDS instances)
-* Higher level abstractions around Kubernetes primitives (eg. a single Resource to define an etcd cluster, backed by a Service and a ReplicationController)
+Responsible for creating, removing and updating [Istio Authorization Policies](https://istio.io/latest/docs/reference/config/security/authorization-policy/#Source) using the [Istio client](https://github.com/istio/client-go) for a given `Profile`.
```
Comment (blairdrummond) [inline README.md:32]: Great! I would add as a small note, atm the only AuthPolicy is to block upload/download from protected-b notebooks

Comment (cboin1996) [reply README.md:32]: Will do

File: README.md [hunk 7]
```diff
@@ -1,176 +1,98 @@
...
+Responsible for the configuration of `Notebook` resources within Kubeflow. This controller adds a default option for running a protected-b `Notebook`.
 
-## Subresources
+### [quotas.go](./cmd/quotas.go)
 
-Custom Resources support `/status` and `/scale` [subresources](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#subresources). The `CustomResourceSubresources` feature is in GA from v1.16.
+Responsible for the generation of `ResourceQuotas` for a given profile. Management of `ResourceQuotas` is essential, as it provides the constraint for total amount of compute resources that are consumable within the `Profile`. Since the `ResourceQuotas` definition included in [quotas.go](./cmd/quotas.go) provides constraints for cpu and memory, the limits for the values must be defined. These limits are defined as `LimitRange` resources and are managed by [limitrange.go](./cmd/limitrange.go).
 
-### Example
+#### Quota Labels
```
Comment (blairdrummond) [inline README.md:67]: Link to this:

https://github.com/StatCan/aaw-kubeflow-profiles-controller/issues/16

Comment (cboin1996) [reply README.md:67]: Do you mean I should replace ####Quota Labels with ####[Quota Labels](https://github.com/StatCan/aaw-kubeflow-profiles-controller/issues/16)?

**Pull request merged**